ansible/roles/vm_set/library/vm_topology.py

```diff
@@ -490,19 +493,33 @@ def get_ovs_br_ports(bridge):
         out = VMTopology.cmd('ovs-vsctl list-ports %s' % bridge)
         return set(out.split('\n'))
 
+
     @staticmethod
-    def get_ovs_port_bindings(bridge):
-        out = VMTopology.cmd('ovs-ofctl show %s' % bridge)
-        lines = out.split('\n')
-        result = {}
-        for line in lines:
-            matched = re.match(r'^\s+(\S+)\((\S+)\):\s+addr:.+$', line)
-            if matched:
-                port_id = matched.group(1)
-                iface_name = matched.group(2)
-                result[iface_name] = port_id
-
-        return result
+    def get_ovs_port_bindings(bridge, vlan_iface = None):
+        # Vlan interface addition may take few secs to reflect in OVS Command,
+        # Let`s retry few times in that case.
+        retries = 0
+        vlan_iface_id = None
+        while retries < RETRIES:
+            out = VMTopology.cmd('ovs-ofctl show %s' % bridge)
+            lines = out.split('\n')
+            result = {}
+            for line in lines:
+                matched = re.match(r'^\s+(\S+)\((\S+)\):\s+addr:.+$', line)
+                if matched:
+                    port_id = matched.group(1)
+                    iface_name = matched.group(2)
+                    result[iface_name] = port_id
+            if vlan_iface is None:
+                return result, False
+            # Check if we have vlan_iface populated
+            vlan_iface_id = result[vlan_iface]
```

Comment:
how can you be sure you have vlan_iface key in the result? I think you need to check that you have vlan_iface key in the result first

Reply:
Sure let me add a check for this.